tracing-subscriber/src/filter/mod.rs

```diff
@@ -118,6 +117,35 @@ impl Filter {
         env::var(env.as_ref())?.parse().map_err(Into::into)
     }
 
+    /// Add a filtering directive to this `Filter`.
```

Comment:
Can you elaborate on what this entails for the filter directives? For instance, would this allow envlogger-style directives + a default filter? How do those filters interact—are they additive, or can they conflict?

Reply:
added to the docs in 74e31bec96de4f43b77711e471be387d9a9ae153 and renamed the method to `add_directive` to try and make it clearer that this is additive.

Reply:
Thanks for clarifying that they're additive. What would be the behavior be if the program is run with `RUST_LOG=trace` and the directive is `info`? Which one would take precedence?

Reply:
the behaviour of adding a new filter that matches _exactly_ the same spans (e.g. in this case, "all spans") actually depends on what `BTreeSet` does, I'll have to check.

Reply:
looks like duplicate insertions are ignored rather than clobbering: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=6e5b25fb71b1c329e7a9ff04d554421f

this seems like the wrong behaviour here, we probably want to overwrite. let me fix that!